subworkflows/local/profiling.nf

```diff
@@ -129,7 +131,37 @@ workflow PROFILING {
         ch_multiqc_files       = ch_multiqc_files.mix( KRAKEN2_KRAKEN2.out.report )
         ch_versions            = ch_versions.mix( KRAKEN2_KRAKEN2.out.versions.first() )
         ch_raw_classifications = ch_raw_classifications.mix( KRAKEN2_KRAKEN2.out.classified_reads_assignment )
-        ch_raw_profiles        = ch_raw_profiles.mix( KRAKEN2_KRAKEN2.out.report )
+        ch_raw_profiles        = ch_raw_profiles.mix(
+            KRAKEN2_KRAKEN2.out.report
+                // Set the tool to be strictly 'kraken2' instead of potentially 'bracken' for downstream use.
+                .map { meta, report -> [meta + [tool: 'kraken2'], report]}
+        )
+
+    }
+
+    if ( params.run_kraken2 && params.run_bracken ) {
+
+        def ch_input_for_bracken = KRAKEN2_KRAKEN2.out.report
+            .filter { meta, report -> meta['tool'] == 'bracken' }
+
+        if (params.kraken2_save_minimizers) {
+            ch_input_for_bracken = KRAKEN2_STANDARD_REPORT(ch_input_for_bracken).report
+        }
+
+        ch_input_for_bracken = ch_input_for_bracken
+            .combine(
```

Comment:
I don't like reusing the channel name here of `ch_input_for_bracken`

Shouldn't the combine be rather a `join` to make sure the kreports are associted with the correct bracken database (based on the bracken database name?)

Reply:
You are completely correct! This will need some more logic to address the problem.